core/src/main/scala/spinal/core/internals/Phase.scala

```diff
@@ -2481,6 +2481,40 @@ class PhaseAllocateNames(pc: PhaseContext) extends PhaseMisc{
     for (c <- sortedComponents.reverse) {
       c.allocateNames(pc.globalScope)
     }
+
+    val invalidVhdlIdentifier = List (
+      ("[^\\w]".r, "contains non-alphanumeric characters"),
+      ("_$".r, "ends with an underscore"),
+      ("__".r, "contains a double underscore"),
+      ("^[^a-zA-Z]".r, "doesn't start with a letter")
+    )
+    val invalidVerilogIdentifier = List (
+      ("[^\\w$]".r, "contains non-alphanumeric or dollar characters"),
+      ("^[^a-zA-Z_]".r, "doesn't start with a letter or underscore")
```

Comment:
> or underscore

Isn't this a VHDL rule ? Starting with underscore is often used in verilog, this for instance allow to make a signal invisible in verilator.

Reply:
Also, if this is moved to the VHDL rullset, a new issue appear that all the _zz_???? generated in the Verilog mode will trigger it as warning :)

Reply:
It tests the _negative_. I.e. the pattern `^[^a-zA-Z_]` matches with any string/name that does not start with either an underscore or an letter. If there is a match, then an error is generated. So the checks pass if there is _no_ match.

So it is at the correct place with the verilog rules.

Reply:
The line `if (!name.startsWith(pc.globalData.anonymSignalPrefix)) {` should prevent any of the automatically generated anonymous signals to trigger warnings or errors... Does it not? 😅

Reply:
Hoooooooooooo right XD
my bad. thanks ^^